CONTRIBUTING.md

```diff
@@ -215,6 +215,25 @@ and false.
 * If a PR includes a deprecation/breaking change, notification should be sent to the
   [envoy-announce](https://groups.google.com/forum/#!forum/envoy-announce) email list.
 
+# Adding new extensions
+
+For developers adding a new extension, one can largely model off of existing extensions.
+
+Extension configuration should be located in a directory structure like
+`api/envoy/extensions/area/plugin/`, for example `api/envoy/extensions/tracers/zipkin/`
+
+The code for the extension should be located under the equivalent
+`source/extensions/area/plugin`, and include an envoy_cc_extension with the
+configuration and tagged with the appropriate security posture, and an
+envoy_cc_library with the code.
```

Comment:
Suggestion: use markup to make Bazel macros distinct, e.g. `*envoy_cc_library*`.